ast/build.rs

```diff
@@ -0,0 +1,130 @@
+use std::{
+    fs::File,
+    io::{Read, Write},
+    path::{Path, PathBuf},
+};
+
+fn main() {
+    println!("cargo:rerun-if-changed=Python.asdl");
+    println!("cargo:rerun-if-changed=asdl_rs.py");
+    // println!("cargo:rerun-if-changed=../scripts/update_asdl.sh");
+
+    let out_dir: PathBuf = std::env::var("OUT_DIR").unwrap().into();
+    // println!("cargo:warning={out_dir:?}");
+    let build_dir = {
+        let mut path = out_dir;
+        path.pop();
+        path.pop();
+        path
+    };
+
+    let def_path = {
+        let mut path = build_dir.clone();
+        path.push("ast_def.rs");
+        path
+    }
+    .to_str()
+    .expect("def path is not str representable")
+    .to_owned();
+    let mod_path = {
+        let mut path = build_dir;
+        path.push("ast_mod.rs");
+        path
+    }
+    .to_str()
+    .expect("mod path is not str representable")
+    .to_owned();
+
+    // // TODO: do this when pyo3 is fixed on apple silicon
```

Comment:
I think shelling out is better than depending on pyo3; it's ideally an uncommon situation and adding a dependency would increase build time for a code path that's rarely used.